lib/Doctrine/Migrations/Tools/TransactionHelper.php

```diff
@@ -36,6 +37,10 @@ private static function inTransaction(Connection $connection): bool
 
         /* Attempt to commit or rollback while no transaction is running
            results in an exception since PHP 8 + pdo_mysql combination */
-        return ! $wrappedConnection instanceof PDO || $wrappedConnection->inTransaction();
+        if (method_exists($wrappedConnection, 'inTransaction')) {
```

Comment:
Only now do I notice that while `PDOConnection` extends `PDO`, it is deprecated in favor of `PDO\Connection` which no longer does in DBAL 3. That means your fix is needed regardless, and that implementing `inTransaction` in `PDO\Connection` will be necessary.

Reply:
True, but this lib does not support yet the newer version of the DBAL so for now this is a non-issue, and this code is future-proof if the method name is the one used in the implementation of doctrine/dbal#4616. A future iteration on this piece of code may of course change the check to use the more reliable `instanceof TransactionAwareDriverConnection` (or whatever name it will have)